File: backend/app/schemas/assistant.py
"""Recruitment assistant API schemas."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import ChatRole


class ConversationCreate(BaseModel):
    title: str | None = Field(default=None, max_length=255)
    candidate_id: uuid.UUID | None = None
    job_id: uuid.UUID | None = None
    application_id: uuid.UUID | None = None


class MessageCreate(BaseModel):
    content: str = Field(min_length=1, max_length=4000)


class AssistantMessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    conversation_id: uuid.UUID
    role: ChatRole
    content: str
    meta: dict[str, Any] | None = None
    created_at: datetime


class ConversationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    title: str
    created_by_user_id: uuid.UUID
    candidate_id: uuid.UUID | None = None
    candidate_name: str | None = None
    job_id: uuid.UUID | None = None
    job_title: str | None = None
    application_id: uuid.UUID | None = None
    created_at: datetime
    updated_at: datetime
    messages: list[AssistantMessageResponse] = Field(default_factory=list)


class ConversationListResponse(BaseModel):
    items: list[ConversationResponse]
    total: int


class ChatReplyResponse(BaseModel):
    conversation: ConversationResponse
    reply: AssistantMessageResponse
    action_result: dict[str, Any] | None = None


class ScheduleInterviewAction(BaseModel):
    type: Literal["schedule_interview"] = "schedule_interview"
    application_id: uuid.UUID | None = None
    scheduled_at: datetime | None = None
    interview_type: str = "video"
    duration_minutes: int = 60
    meeting_link: str | None = None
    location: str | None = None
    send_whatsapp: bool = True

File: frontend/src/types/assistant.ts
export type ChatRole = "user" | "assistant" | "system"

export type AssistantMode = "candidate" | "recruiter" | "admin"

export type AssistantMessage = {
  id: string
  conversation_id: string
  role: ChatRole
  content: string
  meta: Record<string, unknown> | null
  created_at: string
}

export type AssistantConversation = {
  id: string
  title: string
  created_by_user_id: string
  candidate_id: string | null
  candidate_name: string | null
  job_id: string | null
  job_title: string | null
  company_name?: string | null
  application_id: string | null
  created_at: string
  updated_at: string
  messages: AssistantMessage[]
}

export type ConversationListResponse = {
  items: AssistantConversation[]
  total: number
}

export type ChatReplyResponse = {
  conversation: AssistantConversation
  reply: AssistantMessage
  action_result: Record<string, unknown> | null
  follow_ups: string[]
}

export type ConversationCreatePayload = {
  title?: string
  candidate_id?: string
  job_id?: string
  application_id?: string
}

export type AssistantStatus = {
  configured: boolean
  model: string | null
  mode: AssistantMode
  message: string
}
